client/web/web_test.go

```diff
@@ -325,3 +294,133 @@ func TestGetTailscaleBrowserSession(t *testing.T) {
 		})
 	}
 }
+
+// TestAuthorizeRequest tests the s.authorizeRequest function.
+// 2023-10-18: These tests currently cover tailscale auth mode (not platform auth).
+func TestAuthorizeRequest(t *testing.T) {
+	// Create self and remoteNode owned by same user.
+	// See TestGetTailscaleBrowserSession for tests of
+	// browser sessions w/ different users.
+	user := &tailcfg.UserProfile{ID: tailcfg.UserID(1)}
+	self := &ipnstate.PeerStatus{ID: "self", UserID: user.ID}
+	remoteNode := &apitype.WhoIsResponse{Node: &tailcfg.Node{StableID: "node"}, UserProfile: user}
+	remoteIP := "100.100.100.101"
+
+	lal := memnet.Listen("local-tailscaled.sock:80")
+	defer lal.Close()
+	localapi := mockLocalAPI(t,
+		map[string]*apitype.WhoIsResponse{remoteIP: remoteNode},
+		func() *ipnstate.PeerStatus { return self },
```

Comment:
in both uses of mockLocalAPI, you're just passing a self func that returns a static value.  Any reason not to update mockLocalAPI to just take that value?  Did you anticipate needing it to be a function in other contexts?

Reply:
`TestGetTailscaleBrowserSession` actually does update the self pointer on each of the tests (so I could swap between testing a tag-owned self node and a user-owned self node). I'll add a little comment about why it's a function though, is a little confusing.

Reply:
oh, I totally missed that!

Reply:
all good, that wasn't visible in this diff!